filebeat/prospector/log/harvester.go

```diff
@@ -98,9 +111,9 @@ func NewHarvester(
 	}
 
 	// Add outlet signal so harvester can also stop itself
-	outlet.SetSignal(h.done)
-
 	var err error
+
+	outlet = channel.CloseOnSignal(outlet, h.done)
```

Comment:
Comment on line 113 is now in the wrong place. I put this one as early as possible because of our still unsovled race condition.